### PR TITLE
PR:BAEL-6037

### PR DESCRIPTION
**Pull request by BahaaHelmy**
PR to the article named Java Enums with all HTTP status codes [BAEL-6037].

### CODE REVIEW

File: libraries-http-3/pom.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <properties>
+        <maven.compiler.source>20</maven.compiler.source>
+        <maven.compiler.target>20</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <okhttp.version>4.9.2</okhttp.version>
+        <apache.httpclient.version>5.3.1</apache.httpclient.version>
+        <junit.version>4.13.2</junit.version>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:23]: We use JUnit5 for new articles

File: libraries-http-3/pom.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    <parent>
+        <groupId>com.baeldung</groupId>
+        <artifactId>parent-modules</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <properties>
+        <maven.compiler.source>20</maven.compiler.source>
+        <maven.compiler.target>20</maven.compiler.target>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:19]: What is the minimum Java version required? We shouldn't set this higher than the minimum actually required for this article

File: libraries-http-3/pom.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            <artifactId>slf4j-api</artifactId>
+            <version>${slf4j.version}</version>
+        </dependency>
+
+        <!-- SLF4J Binding for Logback -->
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>${logback.version}</version>
+        </dependency>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:71]: These are already defined in the parent module
```suggestion
```

File: libraries-http-3/pom.xml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            <version>${apache.httpclient.version}</version>
+        </dependency>
+
+        <!-- JUnit Dependency -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${junit.version}</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:50]: These are already defined in the parent module
```suggestion
```

File: libraries-http-3/pom.xml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        <okhttp.version>4.9.2</okhttp.version>
+        <apache.httpclient.version>5.3.1</apache.httpclient.version>
+        <junit.version>4.13.2</junit.version>
+        <spring.web.version>5.3.10</spring.web.version>
+        <slf4j.version>1.7.32</slf4j.version>
+        <logback.version>1.2.10</logback.version>
+    </properties>
+
+    <dependencies>
+        <!-- OkHttp Dependency -->
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:30]: Delete these comments, the dependency is clear from the groupId/artifactId already
```suggestion
```

File: libraries-http-3/pom.xml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    </parent>
+
+    <properties>
+        <maven.compiler.source>20</maven.compiler.source>
+        <maven.compiler.target>20</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <okhttp.version>4.9.2</okhttp.version>
+        <apache.httpclient.version>5.3.1</apache.httpclient.version>
+        <junit.version>4.13.2</junit.version>
+        <spring.web.version>5.3.10</spring.web.version>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:24]: Check versions of the libraries and use the latest stable versions

File: pom.xml [hunk 7]
```diff
@@ -652,6 +652,7 @@
                 <module>libraries-data-io</module>
                 <module>libraries-data</module>
                 <module>libraries-files</module>
+	        <module>libraries-http-3</module>
```
Comment (theangrydev) [inline pom.xml:655]: Look for all the places that libraries-http-2 is mentioned and reorder like:
                <module>libraries-http</module>
                <module>libraries-http-2</module>
                <module>libraries-http-3</module>

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: libraries-http-3/src/test/java/com/baeldung/enumwithallhttpstatuscodes/EnumWithHttpStatusCodesLiveTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.junit.Test;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
```
Comment (theangrydev) [inline libraries-http-3/src/test/java/com/baeldung/enumwithallhttpstatuscodes/EnumWithHttpStatusCodesLiveTest.java:16]: Use JUnit5 not JUnit4

File: libraries-http-3/src/test/java/com/baeldung/enumwithallhttpstatuscodes/EnumWithHttpStatusCodesUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
+package com.baeldung.enumwithallhttpstatuscodes;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
```
Comment (theangrydev) [inline libraries-http-3/src/test/java/com/baeldung/enumwithallhttpstatuscodes/EnumWithHttpStatusCodesUnitTest.java:5]: Use JUnit5 not JUnit4

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: libraries-http-3/pom.xml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+         
+    <parent>
+        <groupId>com.baeldung</groupId>
+        <artifactId>parent-modules</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+         
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <okhttp.version>4.10.0</okhttp.version>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:19]: The latest stable version is 4.12.0 https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp

File: libraries-http-3/pom.xml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        <apache.httpclient.version>5.3.1</apache.httpclient.version>
+        <spring.web.version>6.1.5</spring.web.version>
```
Comment (theangrydev) [inline libraries-http-3/pom.xml:21]: The latest stable version is 6.1.11 https://mvnrepository.com/artifact/org.springframework/spring-web

**Review by theangrydev — CHANGES_REQUESTED** [review]

**Review by theangrydev — APPROVED** [review]

**Comment by theangrydev** [comment]
retest this please

**Comment by lor6** [comment]
retest this please

**Pull request merged**